src/cli/Cli.ts

```diff
@@ -158,109 +158,107 @@ export class Cli {
       .then(_ => process.exit(0), err => this.closeWithError(err, optionsWithoutShorts));
   }
 
-  public static executeCommand(command: Command, args: { options: minimist.ParsedArgs }): Promise<void> {
-    return new Promise<void>((resolve: () => void, reject: (error: any) => void): void => {
-      const logger: Logger = {
-        log: (message: any): void => {
-          const output: any = Cli.formatOutput(message, args.options);
-          Cli.log(output);
-        },
-        logRaw: (message: any): void => Cli.log(message),
-        logToStderr: (message: any): void => Cli.error(message)
-      };
-
-      if (args.options.debug) {
-        logger.logToStderr(`Executing command ${command.name} with options ${JSON.stringify(args)}`);
-      }
-
-      // store the current command name, if any and set the name to the name of
-      // the command to execute
-      const cli = Cli.getInstance();
-      const parentCommandName: string | undefined = cli.currentCommandName;
-      cli.currentCommandName = command.getCommandName(cli.currentCommandName);
+  public static async executeCommand(command: Command, args: { options: minimist.ParsedArgs }): Promise<void> {
+    const logger: Logger = {
+      log: (message: any): void => {
+        const output: any = Cli.formatOutput(message, args.options);
+        Cli.log(output);
+      },
+      logRaw: (message: any): void => Cli.log(message),
+      logToStderr: (message: any): void => Cli.error(message)
+    };
 
-      command.action(logger, args as any, (err: any): void => {
-        // restore the original command name
-        cli.currentCommandName = parentCommandName;
+    if (args.options.debug) {
+      logger.logToStderr(`Executing command ${command.name} with options ${JSON.stringify(args)}`);
+    }
 
-        if (err) {
-          return reject(err);
-        }
+    // store the current command name, if any and set the name to the name of
+    // the command to execute
+    const cli = Cli.getInstance();
+    const parentCommandName: string | undefined = cli.currentCommandName;
+    cli.currentCommandName = command.getCommandName(cli.currentCommandName);
 
-        if (args.options.debug || args.options.verbose) {
-          const chalk: typeof Chalk = require('chalk');
-          logger.logToStderr(chalk.green('DONE'));
-        }
+    try {
+      await command.action(logger, args as any);
+    }
+    catch (err: any) {
```

Comment:
I wonder if we can skip this since we're not adding anything to it and simply rethrow the original exception.

Reply:
Seems to be true indeed, this catch looks relatively useless.